internal/storage/sqlindexer_test.go

```diff
@@ -0,0 +1,479 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/elastic/package-registry/internal/database"
+	"github.com/elastic/package-registry/internal/util"
+	"github.com/elastic/package-registry/packages"
+)
+
+func TestSQLInit(t *testing.T) {
+	// given
+	db, err := database.NewMemorySQLDB("main")
+	require.NoError(t, err)
+
+	swapDb, err := database.NewMemorySQLDB("swap")
+	require.NoError(t, err)
+
+	options, err := CreateFakeIndexerOptions(db, swapDb)
+	require.NoError(t, err)
+
+	fs := PrepareFakeServer(t, "testdata/search-index-all-full.json")
+	defer fs.Stop()
+	storageClient := fs.Client()
+
+	ctx := context.Background()
+	indexer := NewIndexer(util.NewTestLogger(), storageClient, options)
+	defer indexer.Close(ctx)
+
+	// when
+	err = indexer.Init(ctx)
+
+	// then
+	require.NoError(t, err)
+}
+
+func BenchmarkSQLInit(b *testing.B) {
+	// given
+	folder := b.TempDir()
+	dbPath := filepath.Join(folder, "test.db")
+	db, err := database.NewFileSQLDB(dbPath)
+	require.NoError(b, err)
+
+	swapDbPath := filepath.Join(folder, "swap_test.db")
+	swapDb, err := database.NewFileSQLDB(swapDbPath)
+	require.NoError(b, err)
+
+	options, err := CreateFakeIndexerOptions(db, swapDb)
+	require.NoError(b, err)
+
+	fs := PrepareFakeServer(b, "testdata/search-index-all-full.json")
+	defer fs.Stop()
+	storageClient := fs.Client()
+
+	logger := util.NewTestLoggerLevel(zapcore.FatalLevel)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		ctx := context.Background()
+
+		indexer := NewIndexer(logger, storageClient, options)
+		defer indexer.Close(ctx)
+
+		err := indexer.Init(ctx)
+		require.NoError(b, err)
```

Comment:
Would it be needed to wrap these lines with a `func()` because of the defer function ? 

```
	for i := 0; i < b.N; i++ {
		func() {
			ctx := context.Background()

			indexer := NewIndexer(logger, storageClient, options)
			defer indexer.Close(ctx)

			err := indexer.Init(ctx)
			require.NoError(b, err)
		}()
	}
```